File: backend/src/main/java/br/ufpr/sept/so2/modules/presenca/domain/EventoEstado.java
package br.ufpr.sept.so2.modules.presenca.domain;

import br.ufpr.sept.so2.shared.domain.exception.DadoInvalidoException;

public enum EventoEstado {
    AGENDADO,
    EM_ANDAMENTO,
    CONCLUIDO;

    public static EventoEstado from(String raw) {
        if (raw == null || raw.isBlank()) {
            throw new DadoInvalidoException("Estado do evento obrigatório.");
        }
        try {
            return EventoEstado.valueOf(raw.trim().toUpperCase());
        } catch (IllegalArgumentException ex) {
            throw new DadoInvalidoException("Estado do evento inválido.");
        }
    }
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/presenca/domain/EventoEstado.kt
package br.ufpr.sept.so2.modules.presenca.domain

import br.ufpr.sept.so2.shared.domain.exception.DadoInvalidoException

enum class EventoEstado {
    AGENDADO,
    EM_ANDAMENTO,
    CONCLUIDO,
    ;

    companion object {
        fun from(raw: String?): EventoEstado {
            if (raw.isNullOrBlank()) {
                throw DadoInvalidoException("Estado do evento obrigatório.")
            }
            return try {
                valueOf(raw.trim().uppercase())
            } catch (_: IllegalArgumentException) {
                throw DadoInvalidoException("Estado do evento inválido.")
            }
        }
    }
}
